protocol-units/settlement/mcr/client/src/eth_client.rs

```diff
@@ -0,0 +1,536 @@
+			self.config.tx_send_nb_retry,
+			self.config.gas_limit,
+		)
+		.await
+	}
+
+	async fn stream_block_commitments(&self) -> Result<CommitmentStream, anyhow::Error> {
+		//register to contract BlockCommitmentSubmitted event
+
+		let contract = MCR::new(self.config.mrc_contract_address.parse()?, &self.ws_provider);
+		let event_filter = contract.BlockAccepted_filter().watch().await?;
+
+		let stream = event_filter.into_stream().map(|event| {
+			event
+				.and_then(|(commitment, _)| {
+					let height = commitment.height.try_into().map_err(
+						|err: alloy::primitives::ruint::FromUintError<u64>| {
+							alloy_sol_types::Error::Other(err.to_string().into())
+						},
+					)?;
+					Ok(BlockCommitment {
+						height,
+						block_id: Id(commitment.blockHash.0),
+						commitment: Commitment(commitment.stateCommitment.0),
+					})
+				})
+				.map_err(|err| McrEthConnectorError::EventNotificationError(err).into())
+		});
+		Ok(Box::pin(stream) as CommitmentStream)
+	}
+
+	async fn get_commitment_at_height(
+		&self,
+		height: u64,
+	) -> Result<Option<BlockCommitment>, anyhow::Error> {
+		let contract = MCR::new(self.config.mrc_contract_address.parse()?, &self.ws_provider);
+		let MCR::getValidatorCommitmentAtBlockHeightReturn { _0: commitment } = contract
+			.getValidatorCommitmentAtBlockHeight(U256::from(height), self.signer_address)
+			.call()
+			.await?;
+		let return_height: u64 = commitment.height.try_into()?;
+		// Commitment with height 0 mean not found
+		Ok((return_height != 0).then_some(BlockCommitment {
+			height: commitment.height.try_into()?,
+			block_id: Id(commitment.blockId.into()),
+			commitment: Commitment(commitment.commitment.into()),
+		}))
+	}
+
+	async fn get_max_tolerable_block_height(&self) -> Result<u64, anyhow::Error> {
+		let contract = MCR::new(self.config.mrc_contract_address.parse()?, &self.ws_provider);
+		let MCR::getMaxTolerableBlockHeightReturn { _0: block_height } =
+			contract.getMaxTolerableBlockHeight().call().await?;
+		let return_height: u64 = block_height.try_into()?;
+		Ok(return_height)
+	}
+}
+
+#[cfg(test)]
+pub mod test {
+	use super::*;
+	use alloy_provider::ProviderBuilder;
+	use alloy_signer_wallet::LocalWallet;
+	use movement_types::Commitment;
+	use std::env;
+
+	// Define 2 validators (signer1 and signer2) with each a little more than 50% of stake.
+	// After genesis ceremonial, 2 validator send the commitment for height 1.
+	// Validator2 send a commitment for height 2 to trigger next epoch and fire event.
+	// Wait the commitment accepted event.
+	//#[ignore]
+	#[tokio::test]
+	async fn test_send_commitment() -> Result<(), anyhow::Error> {
+		//Activate to debug the test.
+		// use tracing_subscriber::EnvFilter;
+
+		// tracing_subscriber::fmt()
+		// 	.with_env_filter(
+		// 		EnvFilter::try_from_default_env().unwrap_or_else(|_| EnvFilter::new("info")),
+		// 	)
+		// 	.init();
+
+		// Inititalize Test variables
+		let rpc_port = env::var("MCR_ANVIL_PORT").unwrap();
```

Comment:
We should pull this out into a config pattern similar to that used elsewhere in the workspace. Save for a future ticket.

Reply:
It uses the var define in mcr-env, it's not the right pattern?